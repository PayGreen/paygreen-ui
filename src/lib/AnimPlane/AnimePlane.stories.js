import React from 'react';
import {
    folder,
    colorThemeOptions,
    colorThemeDefault,
} from '../../shared/constants';
import labels from '../../shared/labels';
import AnimPlaneComponent from './AnimPlane';

export default {
    title: folder.media + 'AnimPlane',
    argTypes: {
        isActive: {
            name: labels.isActive,
            control: 'boolean',
        },
        colorTheme: {
            name: labels.colorTheme,
            options: Object.values(colorThemeOptions),
            control: 'select',
        },
    },
};

export const AnimPlane = args => <AnimPlaneComponent {...args} />;

AnimPlane.args = {
    isActive: false,
    colorTheme: colorThemeDefault,
};
