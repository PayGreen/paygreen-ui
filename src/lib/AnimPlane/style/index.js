import styled from 'styled-components';
import {
    planeAnimationDuration,
    dotsDelay,
    dotsTransitions,
    activeStyle,
} from './base';

const AnimPlaneBase = styled.div`
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;

    .icon {
        position: absolute;
        transform-origin: center;
        transform: scale(0);
        transition: all ${props => props.theme.transition.sm};
    }

    .illu {
        width: 100%;
        max-width: ${props => props.theme.imageSize.sm};
        fill: ${props => props.theme.color[props.colorTheme].main};

        .plane {
            transition: opacity ${props => props.theme.transition.sm};
            transition-delay: ${props => props.dotsCount * dotsDelay}s;
            transform-origin: center;
            animation-fill-mode: forwards;
            animation-delay: ${props => props.dotsCount * dotsDelay}s;
            animation-duration: ${planeAnimationDuration}s;
            animation-timing-function: linear;

            .normal {
                opacity: 0.4;
            }

            .light {
                opacity: 0.2;
            }
        }

        .dots {
            opacity: 0.4;

            path {
                opacity: ${props => (props.isActive ? 0 : 1)};
                transition: all 0;
                ${props => dotsTransitions(props.dotsCount)};
            }
        }
    }

    ${props => (props.isActive ? activeStyle : null)};
`;

export { AnimPlaneBase };
