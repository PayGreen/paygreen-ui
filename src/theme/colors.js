const mainColors = {
    green: '#33ad73',
    blue: '#00a99d',
    orange: '#e58a28',
    pink: '#ec6a6a',
    purple: '#a26ae1',
    red: '#b54025',
};

const colors = {
    green: {
        main: mainColors.green,
        gradientBase: '#77d1a6',
        gradientShade: '#a4db95',
    },
    blue: {
        main: mainColors.blue,
        gradientBase: '#51c6be',
        gradientShade: '#76bbd8',
    },
    orange: {
        main: mainColors.orange,
        gradientBase: '#fab56b',
        gradientShade: '#faa272',
    },
    pink: {
        main: mainColors.pink,
        gradientBase: '#e98e90',
        gradientShade: '#e98eb9',
    },
    purple: {
        main: mainColors.purple,
        gradientBase: '#b887ee',
        gradientShade: '#dd94ef',
    },
};

export { mainColors, colors };
