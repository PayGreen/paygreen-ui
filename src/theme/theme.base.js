const responsive = {
    sm: 550,
    md: 960,
    lg: 1200,
    xl: 1360,
};

const query = (value, operator) => {
    if (operator == 'max') {
        value -= 1;
    }
    return operator + '-width: ' + value + 'px';
};

const pallet = {
    green: {
        main: '#33ad73',
        light: '#d5f3e5',
    },
    blue: {
        main: '#00a99d',
        light: '#e1f3f2',
    },
    orange: {
        main: '#e58a28',
        light: '#fbe6ce',
    },
};

export const ThemeBase = {
    name: 'Base',
    color: {
        primary: {
            ...pallet.green,
            gradientBase: '#77d1a6',
            gradientShade: '#a4db95',
        },
        secondary: {
            ...pallet.blue,
            gradientBase: '#51c6be',
            gradientShade: '#76bbd8',
        },
        tertiary: {
            ...pallet.orange,
            gradientBase: '#fab56b',
            gradientShade: '#faa272',
        },
        quaternary: {
            main: '#ec6a6a',
            light: '#ffe0e1',
            gradientBase: '#e98e90',
            gradientShade: '#e98eb9',
        },
        quinary: {
            main: '#a26ae1',
            light: '#e9d9fb',
            gradientBase: '#b887ee',
            gradientShade: '#dd94ef',
        },
    },
    status: {
        success: {
            ...pallet.green,
        },
        default: {
            ...pallet.blue,
        },
        warning: {
            ...pallet.orange,
        },
        danger: {
            main: '#b54025',
            light: '#f7d7d0',
        },
    },
    font: {
        size: {
            tiny: '1rem',
            xxs: '1.2rem',
            xs: '1.4rem',
            sm: '1.6rem',
            base: '1.8rem',
            md: '2.1rem',
            lg: '3rem',
            xl: '4rem',
        },
        weight: {
            normal: 400,
            bold: 700,
        },
        lineHeight: {
            base: 1,
            md: 1.5,
        },
        spacing: '0.3em',
        underline: {
            space: {
                tiny: '8px',
                xxs: '10px',
                xs: '10px',
                sm: '12px',
                base: '12px',
                md: '12px',
                lg: '24px',
                xl: '24px',
            },
            line: {
                tiny: '2px',
                xxs: '2px',
                xs: '2px',
                sm: '2px',
                base: '2px',
                md: '2px',
                lg: '3px',
                xl: '3px',
            },
            width: {
                tiny: '30px',
                xxs: '40px',
                xs: '40px',
                sm: '50px',
                base: '50px',
                md: '65px',
                lg: '80px',
                xl: '80px',
            },
        },
    },
    line: '2px',
    dash: '3px',
    radius: {
        none: '0',
        sm: '4px',
        md: '8px',
        lg: '12px',
    },
    arrow: {
        sm: '8px',
        md: '12px',
        lg: '16px',
    },
    space: {
        none: '0',
        xs: '4px',
        sm: '8px',
        md: '16px',
        lg: '32px',
        xl: '64px',
    },
    grid: {
        header: '45px',
        headerBig: '70px',
        topbar: '45px',
        sidebar: '380px',
        maxWidth: '1400px',
    },
    blockShift: {
        none: '0',
        xs: '64px',
        sm: '128px',
        md: '256px',
        lg: '320px',
        xl: '384px',
    },
    blockSpace: {
        none: '0',
        xs: '16px',
        sm: '24px',
        md: '40px',
        lg: '52px',
        xl: '68px',
    },
    blockWidth: {
        sm: '400px',
        md: '560px',
        lg: '640px',
        xl: '800px',
    },
    blockHeader: {
        sm: '140px',
        md: '180px',
        lg: '210px',
        xl: '240px',
    },
    bannerHeight: {
        xs: '33vh',
        sm: '50vh',
        lg: '100vh',
    },
    wave: {
        simple: '8vw',
        double: '11vw',
    },
    icon: {
        size: {
            xxs: '12px',
            xs: '16px',
            sm: '20px',
            md: '24px',
            lg: '32px',
            xl: '40px',
        },
        shift: {
            xxs: '2px',
            xs: '3px',
            sm: '4px',
            md: '4px',
            lg: '6px',
            xl: '6px',
        },
    },
    imageSize: {
        auto: 'auto',
        xxs: '150px',
        xs: '240px',
        sm: '400px',
        md: '560px',
        lg: '640px',
        xl: '760px',
    },
    imageCircleSize: {
        auto: 'auto',
        xxs: '38px',
        xs: '60px',
        sm: '80px',
        md: '100px',
        lg: '120px',
        xl: '140px',
    },
    logoHeight: {
        none: 'auto',
        xs: '20px',
        sm: '30px',
        md: '40px',
        lg: '50px',
        xl: '60px',
    },
    logoWidth: {
        none: 'auto',
        xs: '90px',
        sm: '120px',
        md: '150px',
        lg: '200px',
        xl: '250px',
    },
    loaderSize: {
        xxs: '40px',
        xs: '50px',
        sm: '60px',
        md: '70px',
        lg: '80px',
        xl: '90px',
    },
    transition: {
        xxs: '0.1s',
        xs: '0.2s',
        sm: '0.3s',
        md: '0.5s',
        lg: '0.7s',
        xl: '1s',
    },
    zindex: {
        hidden: -100,
        layer: -5,
        background: -1,
        base: 0,
        content: 1,
        corner: 3,
        front: 5,
        loader: 90,
        subHeader: 100,
        menu: 105,
        header: 110,
        popin: 115,
        modal: 120,
    },
    query: {
        min: {
            sm: query(responsive.sm, 'min'),
            md: query(responsive.md, 'min'),
            lg: query(responsive.lg, 'min'),
            xl: query(responsive.xl, 'min'),
        },
        max: {
            sm: query(responsive.sm, 'max'),
            md: query(responsive.md, 'max'),
            lg: query(responsive.lg, 'max'),
            xl: query(responsive.xl, 'max'),
        },
    },
    button: {
        paddingWidth: {
            sm: '22px',
            md: '26px',
            lg: '30px',
        },
        paddingHeight: {
            sm: '12px',
            md: '14px',
            lg: '16px',
        },
        shift: '3px',
        font: {
            sm: '1.2rem',
            md: '1.4rem',
            lg: '1.5rem',
        },
        letterSpacing: {
            sm: '0.3rem',
            md: '0.4rem',
            lg: '0.45rem',
        },
    },
    daButton: {
        buttonHeight: {
            sm: '32px',
            md: '38px',
            lg: '44px',
        },
        shift: {
            sm: '2px',
            md: '4px',
            lg: '6px',
        },
        font: {
            sm: '1.35rem',
            md: '1.4rem',
            lg: '1.7rem',
        },
        iconSpacing: {
            sm: '0.9rem',
            md: '1rem',
            lg: '1.1rem',
        },
    },
    form: {
        inputWidth: {
            xs: '260px',
            sm: '330px',
            md: '400px',
            lg: '500px',
        },
        textareaHeight: '200px',
        radio: {
            md: '120px',
            lg: '150px',
        },
        checkboxSize: {
            sm: '18px',
            md: '20px',
            lg: '22px',
        },
        radioButtonSize: {
            sm: '19px',
            md: '21px',
            lg: '23px',
        },
        radioButtonSpace: {
            sm: '8px',
            md: '10px',
            lg: '10px',
        },
        toggle: '30px',
    },
    corner: {
        size: {
            lg: '34px',
        },
        padding: {
            lg: '6px',
        },
    },
    shadow: {
        size: {
            none: '0 0 0',
            xs: '1px 1px 5px',
            sm: '1px 1px 10px',
            md: '2px 2px 15px',
            lg: '3px 3px 20px',
        },
        opacity: {
            none: 1,
            xs: 0.95,
            sm: 0.9,
            md: 0.8,
            lg: 0.7,
        },
    },
    rotateSize: {
        d0: '0deg',
        d45: '45deg',
        d90: '90deg',
        d135: '135deg',
        d180: '180deg',
        d225: '225deg',
        d270: '270deg',
        d315: '315deg',
    },
    dataBar: {
        height: '10px',
        width: {
            sm: '300px',
            md: '400px',
            lg: '500px',
            xl: '600px',
        },
    },
};
