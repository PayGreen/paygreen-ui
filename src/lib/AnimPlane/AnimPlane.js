import React from 'react';
import PropTypes from 'prop-types';
import {
    colorThemeOptions,
    colorThemeDefault,
    iconSizeOptions,
} from '../../shared/constants';
import { CheckIcon } from '../Icon/Icon';
import { AnimPlaneBase } from './style';
import { illu, dotsCount } from './style/illu';

const AnimPlane = props => {
    return (
        <AnimPlaneBase {...props} dotsCount={dotsCount}>
            <CheckIcon
                colorTheme={props.colorTheme}
                iconSize={iconSizeOptions.xl}
                hasBackground={true}
            />

            {illu}
        </AnimPlaneBase>
    );
};

AnimPlane.propTypes = {
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    isActive: PropTypes.bool,
};

AnimPlane.defaultProps = {
    colorTheme: colorThemeDefault,
    isActive: false,
};

export default AnimPlane;
