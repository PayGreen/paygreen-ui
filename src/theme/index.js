import { ThemeDefault } from './theme.default';
import { ThemeDark } from './theme.dark';

export { ThemeDefault, ThemeDark };
