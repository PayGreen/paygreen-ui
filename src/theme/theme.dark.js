import patternDark from './assets/pattern-dark.png';
import { baseDark, gradients } from './colors';
import { ThemeBase } from './theme.base';

const ThemeDark = {
    ...ThemeBase,
    name: 'Dark',
    color: {
        primary: {
            ...baseDark.green,
            ...gradients.green,
        },
        secondary: {
            ...baseDark.blue,
            ...gradients.blue,
        },
        tertiary: {
            ...baseDark.orange,
            ...gradients.orange,
        },
        quaternary: {
            ...baseDark.pink,
            ...gradients.pink,
        },
        quinary: {
            ...baseDark.purple,
            ...gradients.purple,
        },
    },
    status: {
        success: {
            ...baseDark.green,
        },
        default: {
            ...baseDark.blue,
        },
        warning: {
            ...baseDark.orange,
        },
        danger: {
            ...baseDark.red,
        },
    },
    wab: {
        white00: '#202020',
        white10: '#2c2c2c',
        white20: '#383838',

        grey10: '#494949',
        grey20: '#5b5b5b',
        grey30: '#7f7f7f',
        grey40: '#a2a2a2',
        grey50: '#cfcfcf',
        grey60: '#eeeeee',

        black00: '#ffffff',
    },
    pattern: patternDark,
    logoColor: '#ffffff',
};

export { ThemeDark };
