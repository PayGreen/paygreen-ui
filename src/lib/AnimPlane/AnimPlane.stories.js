import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, radios, boolean } from '@storybook/addon-knobs';
import {
    folder,
    colorThemeOptions,
    colorThemeDefault,
} from '../../shared/constants';
import label from '../../shared/labels';
import AnimPlane from './AnimPlane';

storiesOf(folder.media + 'AnimPlane', module)
    .addDecorator(withKnobs)
    .add('AnimPlane', () => (
        <AnimPlane
            colorTheme={radios(
                label.colorTheme,
                colorThemeOptions,
                colorThemeDefault,
            )}
            isActive={boolean(label.isActive, false)}
        />
    ));
