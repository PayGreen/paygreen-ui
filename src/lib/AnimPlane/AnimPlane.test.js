import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import AnimPlane from './AnimPlane';

it('renders without crashing', () => {
    const component = TestRenderer.create(
        <AnimPlane theme={ThemeDefault} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
});
