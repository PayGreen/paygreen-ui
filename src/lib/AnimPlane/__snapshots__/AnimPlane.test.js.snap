// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="sc-bwzfXH ikYMpd"
>
  <span
    className="sc-bdVaJa bPhrHi icon"
  >
    <svg
      viewBox="0 0 143 143"
    >
      <path
        d="M65.9 122.6c-.9 0-1.7-.3-2.4-.9l-51-47.1c-1.4-1.3-1.5-3.5-.2-4.9s3.5-1.5 4.9-.2l48 44.4L123 28c1.1-1.6 3.3-2 4.9-.9s2 3.3.9 4.9l-60 89c-.6.8-1.5 1.4-2.5 1.5-.1.1-.2.1-.4.1z"
      />
    </svg>
  </span>
  <svg
    className="illu"
    viewBox="0 0 300 215"
  >
    <g
      className="plane"
    >
      <polygon
        className="normal"
        points="296.7,3.2 142.7,78.5 179.2,93.5"
      />
      <polygon
        className="light"
        points="296.7,3.2 179.2,93.5 182.3,132.2"
      />
      <polygon
        className="light"
        points="182.3,132.2 296.7,3.2 194.3,99.6"
      />
      <polygon
        className="normal"
        points="296.7,3.2 194.3,99.6 246,121"
      />
    </g>
    <g
      className="dots"
    >
      <path
        d="M25.5,210.4c-1.3-0.5-2.5-1-3.7-1.6l0.6-1.2c1.1,0.6,2.3,1.1,3.5,1.6L25.5,210.4z"
      />
      <path
        d="M15.4,204c-1.8-1.8-3.4-3.9-4.8-6.2l1.1-0.7c1.3,2.2,2.8,4.2,4.6,5.9L15.4,204z"
      />
      <path
        d="M7.4,190.6c-0.8-2.4-1.4-5-1.7-7.6l1.3-0.2c0.3,2.6,0.9,5.1,1.7,7.4L7.4,190.6z"
      />
      <path
        d="M6.5,175.3H5.2c0-2.6,0.3-5.2,0.7-7.8l1.3,0.2C6.8,170.2,6.6,172.8,6.5,175.3z"
      />
      <path
        d="M8.8,160.2l-1.3-0.4c0.7-2.5,1.6-5,2.6-7.3l1.2,0.5C10.4,155.4,9.5,157.8,8.8,160.2z"
      />
      <path
        d="M14.9,146.4l-1.1-0.7c1.4-2.3,2.9-4.4,4.6-6.3l1,0.9C17.8,142.1,16.3,144.2,14.9,146.4z"
      />
      <path
        d="M24.8,135l-0.8-1c2.1-1.7,4.3-3.1,6.6-4.3l0.6,1.2C28.9,132,26.8,133.4,24.8,135z"
      />
      <path
        d="M38.2,128.2l-0.3-1.3c0.8-0.2,1.7-0.4,2.5-0.5c1.7-0.3,3.5-0.5,5.2-0.6l0.1,1.3c-1.6,0.1-3.3,0.3-5,0.6
 C39.8,127.8,39,128,38.2,128.2z"
      />
      <path
        d="M60.6,129.1c-2.4-0.8-4.9-1.4-7.4-1.7l0.2-1.3c2.5,0.3,5.1,0.9,7.6,1.8L60.6,129.1z"
      />
      <path
        d="M74.1,135.9c-2.3-1.5-4.4-2.8-6.6-3.8l0.6-1.2c2.2,1.1,4.4,2.4,6.7,3.9L74.1,135.9z"
      />
      <path
        d="M86.4,145.1c-2.3-1.9-4.2-3.4-6-4.7l0.8-1c1.8,1.4,3.7,2.9,6.1,4.8L86.4,145.1z"
      />
      <path
        d="M98.3,154.8c-2-1.6-4-3.2-6-4.9l0.8-1c2,1.6,3.9,3.2,6,4.9L98.3,154.8z"
      />
      <path
        d="M110.9,163.9c-2.1-1.3-4.2-2.8-6.4-4.4l0.8-1.1c2.2,1.6,4.3,3,6.3,4.3L110.9,163.9z"
      />
      <path
        d="M124.6,171.1c-2.3-0.9-4.7-2-7-3.3l0.6-1.2c2.3,1.2,4.6,2.3,6.9,3.2L124.6,171.1z"
      />
      <path
        d="M139.6,175.1c-1.8-0.2-3.6-0.6-5.5-1c-0.7-0.2-1.4-0.3-2.1-0.5l0.3-1.3c0.7,0.2,1.4,0.4,2.1,0.5
 c1.8,0.4,3.6,0.8,5.4,1L139.6,175.1z"
      />
      <path
        d="M147.4,175.4l-0.1-1.3c2.6-0.2,5.1-0.6,7.4-1.3l0.4,1.3C152.7,174.8,150.1,175.2,147.4,175.4z"
      />
      <path
        d="M162.2,170.7l-0.7-1.1c2.1-1.3,4-2.9,5.8-4.8l1,0.9C166.4,167.7,164.4,169.4,162.2,170.7z"
      />
      <path
        d="M173,159.6l-1.1-0.7c1.3-2.1,2.4-4.3,3.5-6.8l1.2,0.5C175.5,155.1,174.3,157.4,173,159.6z"
      />
      <path
        d="M179,145.2l-1.3-0.3c0.6-2.3,0.8-3.8,0.8-3.8l1.3,0.2C179.9,141.4,179.6,142.9,179,145.2z"
      />
    </g>
  </svg>
</div>
`;
